code/modules/mob/living/simple_animal/friendly/nian_caterpillar.dm

```diff
@@ -0,0 +1,163 @@
+#define COCOON_WEAVE_DELAY 5 SECONDS
+#define COCOON_EMERGE_DELAY 15 SECONDS
+#define FLYSWATTER_DAMAGE_MULTIPLIER 14
+
+/mob/living/simple_animal/nian_caterpillar
+	name = "nian caterpillar"
+	icon = 'icons/mob/monkey.dmi'
+	icon_state = "mothroach"
+	icon_living = "mothroach"
+	icon_dead = "mothroach_dead"
+	icon_resting = "mothroach_sleep"
+	pass_flags = PASSTABLE | PASSMOB
+	mob_biotypes = MOB_ORGANIC | MOB_BUG
+	mob_size = MOB_SIZE_SMALL
+	ventcrawler = VENTCRAWLER_ALWAYS
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	butcher_results = list(/obj/item/food/snacks/meat = 1)
+	minbodytemp = 0
+
+	blood_color = "#b9ae9c"
+
+	maxHealth = 50
+	health = 50
+	speed = 0.75
+	stop_automated_movement = FALSE
+	turns_per_move = 4
+
+	// What they sound like
+	voice_name = "nian caterpillar"
+	speak_emote = list("flutters", "chitters", "chatters")
+	emote_hear = list("flutters", "chitters", "chatters")
+	emote_see = list("flutters", "chitters", "chatters")
+
+	// Special verbs for when someone interacts with a caterpillar
+	response_help  = "pets"
+	response_disarm = "shoos"
+	response_harm   = "kicks"
+
+	// Xenobiology and cargo are the only ways to get the caterpillar.
+	gold_core_spawnable = FRIENDLY_SPAWN
+
+	// Attack damage #'s
+	melee_damage_lower = 5
+	melee_damage_upper = 8
+	attacktext = "bites"
+	attack_sound = 'sound/weapons/bite.ogg'
+
+	holder_type = /obj/item/holder/nian_caterpillar
+	can_collar = TRUE
+
+	// Transfers antag status, memories, etc.
+	var/datum/mind/origin
+
+	// Evolution
+	var/datum/action/innate/nian_caterpillar/evolve/evolve_action = new()
+	var/nutrition_need = 500 // You need X amount of nutrition / plant matter to cocoon.
+
+/mob/living/simple_animal/nian_caterpillar/Initialize()
+	. = ..()
+	real_name = name
+	add_language("Tkachi")
+	evolve_action.Grant(src)
+
+/datum/action/innate/nian_caterpillar/evolve
```

Comment:
put this datum action below all the caterpillar code please, so its nice and organized into
caterpillar code
evolve datum/action code